### ISSUE TITLE
Tried to alter Replay to run with 10 Bots - Did not work

### ISSUE THREAD

**Issue by PlamAtom** [splewis/csgo-practice-mode#220]
I want to rebuild demos using Practicemode because there is to my knowledge no free demo replay plugin.

### Expected behavior

executes role 0-9 (10 Roles) (Yes and i renamed the recordings)

### Actual behavior

executes role 0-4(5 Roles)

### Steps to reproduce
- Plugin version: 1.3.3
- Sourcemod version: 1.1.1
- Steps to reproduce (please be specific): Altered accessable code so might be impossible without editing the sourcepawn code. 

"Replays"
{
	"0"
	{
		"name"		"unnamed - use .namereplay on me!"
		"role1"
		{
			"file"		"addons\sourcemod\data\botmimic\practicemode/de_inferno/role0/a.rec"
			"team"		"CT"
		}
		"role2"
		{
			"file"		"addons\sourcemod\data\botmimic\practicemode/de_inferno/role1/b.rec"
			"team"		"CT"
		}
		"role3"
		{
			"file"		"addons\sourcemod\data\botmimic\practicemode/de_inferno/role2/c.rec"
			"team"		"CT"
		}
		"role4"
		{
			"file"		"addons\sourcemod\data\botmimic\practicemode/de_inferno/role3/d.rec"
			"team"		"CT"
		}
		"role5"
		{
			"file"		"addons\sourcemod\data\botmimic\practicemode/de_inferno/role4/e.rec"
			"team"		"CT"
		}
		"role6"
		{
			"file"		"addons\sourcemod\data\botmimic\practicemode/de_inferno/role5/a-2.rec"
			"team"		"T"
		}
		"role7"
		{
			"file"		"addons\sourcemod\data\botmimic\practicemode/de_inferno/role6/b-2.rec"
			"team"		"T"
		}
		"role8"
		{
			"file"		"addons\sourcemod\data\botmimic\practicemode/de_inferno/role7/c-2.rec"
			"team"		"T"
		}
		"role9"
		{
			"file"		"addons\sourcemod\data\botmimic\practicemode/de_inferno/role8/d-2.rec"
			"team"		"T"
		}
		"role10"
		{
			"file"		"addons\sourcemod\data\botmimic\practicemode/de_inferno/role9/e-2.rec"
			"team"		"T"
		}
	}
}

**Comment by splewis** [comment]
The code was written only to support 5 replay clients at once.

If you really want to try to make this work with 10 clients, you can recompile modifying the `MAX_REPLAY_CLIENTS ` value:
https://github.com/splewis/csgo-practice-mode/blob/master/scripting/practicemode/botreplay.sp#L4

The bot replay behavior is, in general, somewhat buggy. So I'd be somewhat hesitant to make that change by default (and make the menus more difficult to navigate than they already are).

**Comment by PlamAtom** [comment]
Can i use SourcePawn Compiler GUI or what compiler did you use? Because i can guess that the popular SourcePawn Editor has also a compiler.

**Comment by PlamAtom** [comment]
Hey you do mind telling me which encoding the ".rec" files have in which the replay is saved? because i might want to look at the data.

**Comment by splewis** [comment]
I tweaked the code (https://github.com/splewis/csgo-practice-mode/commit/375348ce2924c529356fe1b7eca88b2d9e0630bd) then reverted it (https://github.com/splewis/csgo-practice-mode/commit/1e5cb8ff4d1f4c286dfbff09a4959a98729e7d7f), so you can download an automatic build with MAX_REPLAY_CLIENTS set to 10 from https://ci.splewis.net/job/csgo-practice-mode/311/.

**Comment by splewis** [comment]
> Hey you do mind telling me which encoding the ".rec" files have in which the replay is saved? because i might want to look at the data.

All of that detail is from the botmimic plugin itself, not practicemode, but from a quick look, I see:
* A header: https://github.com/peace-maker/botmimic/blob/master/scripting/botmimic.sp#L965-L993
* A frame format: https://github.com/peace-maker/botmimic/blob/master/scripting/botmimic.sp#L350-L451

**Comment by PlamAtom** [comment]
Is the Menu also from the botmimic plugin because i now want to change the the bot menu to allow me to record role 6-10. I don't want to bother you so much with this ^^

**Comment by splewis** [comment]
No, all the menus are practicemode, botmimic is just providing the natives to start a recording to <file x> and replay a recording from <file x>.

**Comment by PlamAtom** [comment]
I think i found it. csgo-practice-mode/scripting/practicemode/botreplay_editor.sp 
....
/* Page 1 */
  for (int i = 0; i < MAX_REPLAY_CLIENTS; i++) {
    bool recordedLastRole = true;
    if (i > 0) {
      recordedLastRole = HasRoleRecorded(g_ReplayId[client], i - 1);
    }
    int style = EnabledIf(recordedLastRole);
    if (HasRoleRecorded(g_ReplayId[client], i)) {
      char roleName[REPLAY_NAME_LENGTH];
      if (GetRoleName(g_ReplayId[client], i, roleName, sizeof(roleName))) {
        AddMenuIntStyle(menu, i, style, "Change player %d role (%s)", i + 1, roleName);
      } else {
        AddMenuIntStyle(menu, i, style, "Change player %d role", i + 1);
      }
    } else {
      AddMenuIntStyle(menu, i, style, "Add player %d role", i + 1);
    }
  }
...

But this would mean it should work because i is dependet on

**Comment by PlamAtom** [comment]
I checked again and it appears that this is not working at all. I dunno how but it only played 3 more roles from my custom replay in the first place i guess because i was doing it while the server was on or something. Meanwhile i can only again spawn 5 bots, i tried using your 311 but that did not work either (I just copy&pasted Scripting). Only CTs were spawning, i tried making all CTs but that did not work either...

**Comment by PlamAtom** [comment]
Hey, I tried the whole thing again with my custom 10 player replay but it doesn't seem to work only the first 5 bots (CTs) appear. 

I first thought it worked because a bot died on the enemy side at the beginning of the replay. Because for some reason some roles were deleted I thought it was a minor glitch.

**Comment by PlamAtom** [comment]
I think it's just a little more complicated... thanks for your help I'll look at the code and see if i find any hint. Very nice of you to help me. Maybe it's just not meant to be....

**Comment by PlamAtom** [comment]
Solution:
The server must be reinstalled with the Practicemode 311
The start option "-maxplayers_override 12" must be set because default is 10 which means 9 bots plus the one player, who does even matter in .spec 

Thank you very much for your help. I noticed that the bots fly away quite funny when you end the replay with "Noclip".

**Comment by mariuzkve** [comment]
Where can I find Practicemode 311 build? Link seems to be down.
